src/library/scala/collection/MapView.scala

```diff
@@ -21,23 +24,35 @@ trait MapView[K, +V]
 
   override def view: MapView[K, V] = this
 
+  def concat[V1 >: V](that: SomeMapOps[K, V1]): MapView[K, V1] = mapFactory.from(new MapView.Concat(this, that))
```

Comment:
Is there any need to go through `mapFactory.from` in all these methods instead of returning the View directly?

Reply:
I figured I would follow convention I found in other collections, which use the `iterableFactory` to produce the final return value. The benefit being that classes extending mapView only need to change their mapFactory, to change the implementation of all these methods. But not strictly required for this PR, you're right.